### PR TITLE
feat(color): Add on-surface and surface to theme. 

### PR DESCRIPTION
**Pull request by williamernest**
Fixes: #2482

### CODE REVIEW

File: packages/mdc-card/mdc-card.scss [hunk 1]
```diff
@@ -23,15 +23,15 @@
...
+  @include mdc-card-fill-color(surface);
   @include mdc-card-corner-radius(2px);
   @include mdc-elevation(2);
   @include mdc-card-container-layout_;
 }
 
 .mdc-card--stroked {
   @include mdc-elevation(0);
-  @include mdc-card-stroke(#dbdbdb);
+  @include mdc-card-stroke(rgba(mdc-theme-prop-value(on-surface), .12));
```
Comment (bonniezhou) [inline packages/mdc-card/mdc-card.scss]: This will make the stroke translucent, use the Sass `mix()` function instead:
```
mix(mdc-theme-prop-value(on-surface), mdc-theme-prop-value(surface), 12%)
```
Note: the `mix()` function takes percentages so make sure to use 12%, not .12

File: packages/mdc-chips/_variables.scss [hunk 2]
```diff
@@ -14,16 +14,18 @@
 // limitations under the License.
 //
 
+@import "@material/theme/variables";
+
 $mdc-chip-border-radius-default: 16px;
-$mdc-chip-fill-color-default: rgba(black, .08);
-$mdc-chip-ink-color-default: rgba(black, .6);
+$mdc-chip-fill-color-default: rgba(mdc-theme-prop-value(on-surface), .12);
```
Comment (bonniezhou) [inline packages/mdc-chips/_variables.scss]: These should all use the `mix()` function too, but that can go in another PR (to fix #2479)

File: packages/mdc-theme/mdc-theme.scss [hunk 3]
```diff
@@ -24,7 +24,7 @@
 
 // Special case, so that .mdc-theme--background changes background color, not text color.
 .mdc-theme--background {
```
Comment (bonniezhou) [inline packages/mdc-theme/mdc-theme.scss]: This doesn't look like it's being used or documented anywhere so we can probably get rid of it?

Comment (williamernest) [reply packages/mdc-theme/mdc-theme.scss]: We're keeping background. I reverted changes that I made to the components other than chips/card.

File: packages/mdc-theme/_variables.scss [hunk 4]
```diff
@@ -32,6 +32,9 @@ $mdc-theme-secondary: $mdc-theme-accent !default;
 $mdc-theme-on-secondary: if(mdc-theme-contrast-tone($mdc-theme-secondary) == "dark", #000, #fff) !default;
 $mdc-theme-background: #fff !default; // White
 
+$mdc-theme-surface: white !default;
```
Comment (kfranqueiro) [inline packages/mdc-theme/_variables.scss]: Nit: change to #fff since we changed all the others to #fff

File: packages/mdc-card/mdc-card.scss [hunk 5]
```diff
@@ -24,15 +24,15 @@
...
+  @include mdc-card-fill-color(surface);
   @include mdc-card-corner-radius(2px);
   @include mdc-elevation(2);
   @include mdc-card-container-layout_;
 }
 
 .mdc-card--stroked {
   @include mdc-elevation(0);
-  @include mdc-card-stroke($mdc-card-stroke-color);
+  @include mdc-card-stroke(mix(mdc-theme-prop-value(on-surface), mdc-theme-prop-value(surface), 12%));
```
Comment (kfranqueiro) [inline packages/mdc-card/mdc-card.scss]: Can we update the variable instead of replacing it here? (IIRC I added that variable for reuse with shape, at least in demos, but potentially of use to users for similar purposes as well)

Also FYI this results in changing from `#dbdbdb` to `#e0e0e0`
